server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java

```diff
@@ -54,28 +59,55 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         TaskFrameworkProperties taskFrameworkProperties = getConfiguration().getTaskFrameworkProperties();
         int size = taskFrameworkProperties.getSingleFetchCheckHeartTimeoutJobRows();
         int heartTimeoutPeriod = taskFrameworkProperties.getJobHeartTimeoutSeconds();
+        handleGeneralHeartTimeoutJobs(size, heartTimeoutPeriod);
+        if (taskFrameworkProperties.getRunMode().isProcess()) {
+            handleNotLocalAndProcessIsNotExitsJobs(size, heartTimeoutPeriod);
+        }
+    }
+
+    private void handleGeneralHeartTimeoutJobs(int size, int heartTimeoutPeriod) {
         // find heart timeout job
         Page<JobEntity> jobs = getConfiguration().getTaskFrameworkService()
                 .findHeartTimeTimeoutJobs(heartTimeoutPeriod, 0, size);
-        jobs.forEach(this::handleJobRetryingOrCanceled);
+        jobs.forEach(j -> handleJobRetryingOrFailed(j, a -> {
+            try {
+                // destroy executor
+                getConfiguration().getJobDispatcher().destroy(JobIdentity.of(a.getId()));
+            } catch (JobException e) {
+                throw new TaskRuntimeException(e);
+            }
+        }));
+    }
 
+    private void handleNotLocalAndProcessIsNotExitsJobs(int size, int heartTimeoutPeriod) {
+        // docker restart then ip has been changed and process been destroyed,
+        // so we should find heart timeout not running in local when process mode
+        Page<JobEntity> jobs = getConfiguration().getTaskFrameworkService()
+                .findHeartTimeTimeoutNoLocalJobs(heartTimeoutPeriod, 0, size);
+        jobs.stream().filter(a -> {
+            ExecutorIdentifier ei = ExecutorIdentifierParser.parser(a.getExecutorIdentifier());
+            return !(HttpUtil.isConnectable(ei.getHost(), ei.getPort(), 3));
+        }).forEach(j -> handleJobRetryingOrFailed(j, a -> {
```

Comment:
throw exception in forEach iterator may leads unexpected behavior

Reply:
ok, add try catch in `handleJobRetryingOrFailed`  to handle exception